src/file_io.jl

```diff
@@ -66,6 +81,8 @@ save(path::String, args...; kwargs...) = open(io -> save(io, args...; kwargs...)
 
 Load the JLSOFile from the io and deserialize the specified objects.
 If no object names are specified then all objects in the file are returned.
+
+Warning: This method will return `Dict{Symbol, Any}` in the next major release.
```

Comment:
What we could do is implement a custom `AbstractDict{Symbol}` subtype,
that overloads `getindex(::CustomDict, ::String)`
to give a deprecation warning and then to convert the `String` to a `Symbol`.

If we want to be mild evil we could just do it to `Dict{Symbol}` with a little type-piracy.
It is a fairly minor case of type-piracy since it is turning a error into a nonerror.

Reply:
I think returning a different `Dict` type is potentially more breaking (harder to debug) and I think the type-piracy on `Dict{Symbol}` could be a bit confusing if an error in completely unrelated code would throw a JLSO deprecation warning, but it's a good idea.

Reply:
Legit